router/h2/src/main/scala/io/buoyant/router/h2/ClassifiedRetryFilter.scala

```diff
@@ -137,6 +138,12 @@ class ClassifiedRetryFilter(
         case e if responseClassifier(H2ReqRep(req, Throw(e))).contains(ResponseClass.RetryableFailure) =>
           // Request is retryable, attempt to create a new child request stream.
           retry(orElse = { retriesStat.add(count); Future.exception(e) })
+        case NonFatal(e) =>
+          // request is not retryable, we need to make sure the buffer is discarded
```

Comment:
I am not 100% sure, but is this intended to be in this PR? It could be that the base commit is still working off of another branch. Doing a rebase with current master will reduce the diff needed to review.